libcxx/include/__type_traits/remove_pointer.h

```diff
@@ -24,7 +24,7 @@ struct remove_pointer {
 };
 
 template <class _Tp>
-using __remove_pointer_t = __remove_pointer(_Tp);
+using __remove_pointer_t = typename remove_pointer<_Tp>::type;
```

Comment:
Did you find that GCC's `__remove_pointer` was fine when nested within a class template but problematic when used directly in an alias template?

Can we just change the condition of this branch to exclude GCC?

Reply:
Hello frederick,

I think there are many similar new builtins implemented in furture gcc releases which might cause the similar issues.

So you prefer to
```
diff --git a/libcxx/include/__type_traits/remove_pointer.h b/libcxx/include/__type_traits/remove_pointer.h
index 54390a1939f7..0bd0fbf4485d 100644
--- a/libcxx/include/__type_traits/remove_pointer.h
+++ b/libcxx/include/__type_traits/remove_pointer.h
@@ -17,7 +17,7 @@

 _LIBCPP_BEGIN_NAMESPACE_STD

-#if !defined(_LIBCPP_WORKAROUND_OBJCXX_COMPILER_INTRINSICS) && __has_builtin(__remove_pointer)
+#if !defined(_LIBCPP_WORKAROUND_OBJCXX_COMPILER_INTRINSICS) && __has_builtin(__remove_pointer) && !defined(_LIBCPP_COMPILER_GCC)
 template <class _Tp>
 struct remove_pointer {
   using type _LIBCPP_NODEBUG = __remove_pointer(_Tp);
```

Right?

Yours,

Reply:
Here is a complete list of the newly added gcc builtin functions

https://github.com/llvm/llvm-project/issues/91831#issuecomment-2105592054.

Maybe we can add these macro guards as a whole, or we can just add one for remove_pointer builtin function for now.